parsec/mca/device/cuda/device_cuda_component.c

```diff
@@ -192,9 +192,9 @@ static int device_cuda_component_register(void)
     (void)parsec_mca_param_reg_int_name("device_cuda", "max_streams",
                                         "Maximum number of Streams to use for the GPU engine; 2 streams are used for communication between host and device, so the minimum is 3",
                                         false, false, PARSEC_GPU_MAX_STREAMS, &parsec_cuda_max_streams);
-    (void)parsec_mca_param_reg_int_name("device_cuda", "sort_pending_tasks",
```

Comment:
do this make sense to exist per devide or be generic and apply to all devices equally ?

Reply:
The sort function itself has moved to the generic device_gpu code, so it makes sense that the control for it would be at the same framework level.

For the more general question, should it apply per device type, no serious machine is having mixed device types, so its a low-importance feature. Should it be per-device (not by type, by devindex), maybe, but that would be a large refactor without proofs that it serves a purpose atm.

Reply:
There is a runtime with serious funding, and application integration perspectives, that runs on exactly this "benefit". Why restricting the capability of parsec when it can easily be enriched.

Moreover, the sort function can be common, it is just the strategy that is device dependent.

Reply:
See commit b098633e7 if it implements the preferred approach?